lib/Parser/RegexRuntime.h

```diff
@@ -851,39 +840,44 @@ namespace UnifiedRegex
     // Built-in assertions
     //
 
-    struct BOITestInst : Inst, HardFailMixin
+    // BOI = Beginning of Input
+    template <bool canHardFail>
+    struct BOITestInst : Inst
     {
-        inline BOITestInst(bool canHardFail) : Inst(BOITest), HardFailMixin(canHardFail) {}
+        inline BOITestInst();
```

Comment:
nit: I don't think this needs to be declared as inline if you're putting the implementation in the cpp.

Reply:
I think it's helpful for the reader to know this is defined elsewhere (at least that's how I read this).

Reply:
Turns out this declaration is also required or the compiler assumes it was deleted and explicit template overrides conflict with that.

`note: 'UnifiedRegex::BOITestInst<true>::BOITestInst(void)': function was implicitly deleted because a base class 'UnifiedRegex::Inst' has either no appropriate default constructor or overload resolution was ambiguous`